gaphor/SysML/blocks/block.py

```diff
@@ -103,7 +103,8 @@ def update_shapes(self, event=None):
                 and [
                     self.block_compartment(
                         self.diagram.gettext("parts"),
-                        lambda a: not a.association and a.aggregation == "composite",
+                        lambda a: a.aggregation
+                        and a.aggregation in ["composite", "shared"],
```

Comment:
Nice catch!

For those cases I ted to use tuples instead of lists.

```suggestion
                        and a.aggregation in ("composite", "shared"),
```